Makes plasmamen incompatible with DNA scanners and makes them stop burning after a while.

Shadowlight213 opened the pull request:
Plasmamen now will stop igniting after taking enough damage or being dead long enough.
The DNA and cloning scanner now rejects plasmamen DNA.
Plasmamen are restricted from geneticist roles as they are incompatible with all the machines there.
Fixes a runtime whenputting plasmamen in the gibber.
:cl:
rscdel: Plasmamen are no longer compatible with the DNA or cloning scanners.
rscdel: Plasmamen can no longer be geneticists.
bugfix: Plasmamen now stop igniting while dead after taking enough damage or being dead long enough.
/:cl:

Shadowlight213 committed:
Makes Plasmamen dust on death. Also adds a plasmaman meat and fixes a runtime from gibbing them in the gibber.

Buggy123 commented:
IMO you should just make them burn when cloning again, so that if you actually make a plasma cloning chamber they can be cloned.

Cheimon commented:
:+1: Eternal fire sources are bad, more 'hardmode' to the plasma race is good, and it makes sense too that a single cloning machine wouldn't work for all species.

Buggy123 commented:
Maybe make it so that dead and burning plasmamen dust?

SweaterKittens commented:
First off, I didn't think anyone changed them so they didn't burn while cloning. They should still burn while cloning, that was part of them being a hardmode species. 

Regarding dusting on death - Please no. Without being clonable you have to resort to some pretty complicated tactics to get the player back in the round already, usually human'ing a monkey and doing some brain transplanting. As long as them bursting into flames up on being cloned is fixed, there should be no need to make them dust upon death.

Regarding them being eternal fire sources after death - This was a concern many people had when I was doing initial feedback, and my solution was to have them "burn out" after being dead for a certain amount of time (i.e., not defibbable, no chance they're coming back, at this point they're just a fire hazard), and the flames would stop. However, the coder I was sponsoring abandoned the project during the last stages of bugfixing, so element was never added. It would be great if someone else wanted to add these things.

When you die as a Plasmaman, you're already pretty assuredly out of the round (Especially if them being able to be cloned gets fixed, I don't know why that was added). Dusting them is a little excessive. As the player who balanced and was in charge of getting Plasmamen /tg/ ready, I'd be really happy to talk to anyone who wants to discuss their balance and potential changes.

KazeEspada commented:
I just took the plasmameme to robotics and told the roboticist "Put on his jumpsuit during brain removal."

Shadowlight213 committed:
Removed Dusting
Plasmamen now burn out after some time dead or if enough damage is taken.
Plasmamen are now incompatible with DNA and cloning scanners
Plasmamen are unable to pick the geneticist job

KorPhaeron commented on code/game/machinery/computer/dna_console.dm:
```diff
@@ -528,7 +528,7 @@
 	if(connected)
 		viable_occupant = connected.occupant
-		if(!istype(viable_occupant) || !viable_occupant.dna || (viable_occupant.disabilities & NOCLONE))
+		if(!istype(viable_occupant) || !viable_occupant.dna || (viable_occupant.disabilities & NOCLONE) || (viable_occupant.dna.species.id == "plasmaman"))
```
Snowflake as fuck. Is there not a NOCLONE flag for species?

Shadowlight213 replied:
NOCLONE is what husks use iirc. Making a species flag version might be a good idea, as it would also be needed for synths as cloning them breaks their disguises.

KorPhaeron commented on code/game/machinery/computer/cloning.dm:
```diff
@@ -369,6 +369,9 @@
+	if (subject.dna.species.id == "plasmaman")
+		scantemp = "<font class='bad'>Subject genetic structure incompatible. Cloning of this species impossible.</font>"
+		return
 
```
Snowflake

duncathan replied:
Yea ideally there'd be a species flag to check instead of this

SweaterKittens commented:
Awesome, good stuff Shadow! :+1:

KorPhaeron commented:
@Shadowlight213 

I'll merge it once you fix the snowflake

KorPhaeron commented:
@Shadowlight213 

Fix this instead of coding for goon you traitor

Cheridan commented:
I agree with @Buggy123. I've seen a clever plasmaman engineer create a plasma-filled cloning room with a disposal tube entrance. It was pretty great. I don't think we should take away that possibility

SweaterKittens commented:
@Cheridan I saw that the other day too, it was /very/ cool, and took him all round to set up. If someone wants to go to that much time and effort to set up something special like that, I think that's great. However, returning things to the way there were before where Plasmamen burst into flames in cloning, would make the old method of "clone until they pop out dead, suit them up then defib them under showers" work again. I have no personal problems with this, as it's a good bit of trouble to go to and takes careful observation, but I'd like to make everyone happy.

Shadowlight213 commented:
Hmm. I could try that I suppose. Although the thing is, they can still be brought back into the round. Just not as the plasmaman race.
I'll probably switch the genetics scanner exception to the rad_immune species flag.
Do defibs seriously not count clone damage? That seems odd, considering clone damage would be as damaging to a body as burn or brute.
The reason I didn't simply make them burn again, is I can't find the damn commit where it was changed. If I did simply readd the burning, I'd also want to make the defibs not work with heavy clone damage. If you want to clone your plasma bros, you need to build a plasma room.
Alternative option could actually be to make them only scannable while alive, so if they want to be cloned they need to go back themselves up.

@KorPhaeron reeeeeeeee. Fine I'll do it tonight if I remember.

Shadowlight213 commented on code/game/machinery/computer/dna_console.dm:
```diff
@@ -85,7 +85,7 @@
 		if(connected.occupant)	//set occupant_status message
 			viable_occupant = connected.occupant
-			if(viable_occupant.has_dna() && (!(viable_occupant.disabilities & NOCLONE) || (connected.scan_level == 3)))	//occupent is viable for dna modification
+			if(viable_occupant.has_dna() && !(viable_occupant.dna.species.id == "plasmaman") && (!(viable_occupant.disabilities & NOCLONE) || (connected.scan_level == 3)))	//occupent is viable for dna modification
```
Gonna change this to RADIMMUNE flag

Shadowlight213 commented:
I honestly am not motivated to finish this up, and rather than Have it sit for eternity, I'm gonna close this and reopen it if I feel like fixing it.
